### DISCUSSION TITLE
beforeAll hook is not working.

### DISCUSSION THREAD

**Discussion by AutoQA23** [akshayp7/playwright-typescript-playwright-test#55] [Q&A] [answered]
Hi Akshay, 

beforeAll hook is not working in your framework. It is igmoring the test itself. why?

test.beforeAll(async () => {
  console.log('Executing beforeAll hook');
});

**Comment by akshayp7** [comment, marked as answer]
Hi thank you for your patience. Have you tried 'global-setup.ts' file in framework. All basic setups are done here .

**Comment by jagdishgubbi** [comment]
Hi Akshay,
I was able to run one of the test successfully, but do not see any html report and PDF report generated under target folder. I did manually create folders with names "HTMLReport" and "PdfReport" under target folder. Appreciate if your help

**Reply by akshayp7** [reply]
@jagdishgubbi : in playwright.config.ts modify this line with your custom location . Pdf reports are not implemented. You can only read pdf files in my framework.

reporter: [[`./CustomReporterConfig.ts`], [`allure-playwright`], [`html`, { outputFolder: 'html-report', open: 'never' }]],

**Reply by jagdishgubbi** [reply]
Thanks for your response Akshay. I am using playwrite with Java Framework. 
https://github.com/akshayp7/playwright-java-cucumber.git

I was actually running the feature file "Run as->Cucumber feature" from eclipse. After running "Testrunner" as Jnuit tests I was able to see the PDF and HTML reports being generated.

Is there an way I can generate these reports (html and PDF) when I run single scenario or one single feature file. ?

**Reply by akshayp7** [reply]
Can you try tagging single test cases using cucumber tag and run maven command like 
mvn test -Dcucumber.filter.tags="@debug1"

**Reply by JagadishGubbiVM** [reply]
@akshayp7 This worked. Thank you very much.
One last request. Is there a plan to include TestNG to this Playwright+Java+BDD framework where we can configure parallel executions, browser type , Tags etc in TestNG.xml file. (Where you have used Junit currently).

**Reply by akshayp7** [reply]
It can be achieved using Junit as well
For executing test cases in parallel, navigate to pom.xml and in plugin in configuration section provide <skipTests>false</skipTests> Now if you want to run 4 features in parallel provide <threadCount>4</threadCount> in maven-failsafe-plugin section in pom.xml and execute below command
mvn verify
